File: framework/languages/dotnet/e2e/RegistryMessaging/Shared/Messages.cs
namespace RegistryMessaging.Shared;

public sealed record ProfileReq(string Value);

public sealed record ProfileRes(string Value, string ProviderRid);

public sealed record DynamicProfileReq(string RegistryRouterEndpoint, string Value, int ExpectedReady = 1);

public sealed record DynamicProfileBatchReq(
    string RegistryRouterEndpoint,
    string[] Values,
    int ExpectedReady = 1);

public sealed record ProfileMsg(string CommandId);

public sealed record EvidenceWaitReq(string Contains, int TimeoutMilliseconds = 10000);

public sealed record PayloadReq(string Marker, string Payload);

public sealed record PayloadRes(string Marker, int Length, string Sha256);

public sealed record WorkflowReq(string Value);

public sealed record WorkflowRes(string Value, string ProviderRid);

public sealed record ScenarioRoutePing(string Value);

public sealed record ScenarioRoutePong(
    string Value,
    string ProviderRid,
    string SourceRid);

public sealed record RouteMissingRes(bool Failed);

public sealed record RequestFailureRes(bool Failed, string FailureType);

File: framework/languages/node/e2e/RegistryMessaging/Shared/messages.ts
import { createHash } from 'node:crypto';

export const PacketNames = {
  profileRequest: 'ProfileRequest',
  profileCommand: 'ProfileCommand',
  payloadRequest: 'PayloadRequest',
  workflowRequest: 'WorkflowRequest',
  scenarioRoutePing: 'ScenarioRoutePing',
  missingProfileRequest: 'MissingProfileRequest',
  missingProfileCommand: 'MissingProfileCommand'
} as const;

export interface ProfileRequest {
  readonly value: string;
}

export interface ProfileReply {
  readonly value: string;
  readonly providerRid: string;
}

export interface ProfileCommand {
  readonly commandId: string;
}

export interface EvidenceWaitRequest {
  readonly contains: string;
  readonly timeoutMilliseconds?: number;
}

export interface PayloadRequest {
  readonly marker: string;
  readonly payload: string;
}

export interface PayloadReply {
  readonly marker: string;
  readonly length: number;
  readonly sha256: string;
}

export interface WorkflowRequest {
  readonly value: string;
}

export interface WorkflowReply {
  readonly value: string;
  readonly providerRid: string;
}

export interface ScenarioRoutePing {
  readonly value: string;
}

export interface ScenarioRoutePong {
  readonly value: string;
  readonly providerRid: string;
  readonly sourceRid: string;
}

export interface RouteMissingResult {
  readonly failed: boolean;
}

export interface RequestFailureResult {
  readonly failed: boolean;
  readonly failureType: string;
}

export function sha256Hex(value: string): string {
  return createHash('sha256').update(value, 'utf8').digest('hex').toUpperCase();
}
